docs/en/writing-running-appium/android_coverage.md

```diff
@@ -0,0 +1,292 @@
+# Android Coverage Guide
+## Requirement
+* Only Emulators or rooted phones are acceptable
+* Need to add UiAutomator2 support in your apps. That is to say, you need to implement a subclass of `Instrumentation`. `Instrumentation` will be responsible to collect your coverage. 
+* Need to implement a BroadcastReceiver to export coverage to disk files. This is because `Instrumentation` only collects data into memory.
+
+## Project Structure
+You may need the following structure of files:
+```
+src/main/java/com/example/pkg
+   |____ MainActivity.java *Your main activity*
+   |____ InstrumentActivityListener.java *A customized interface for exporting coverage to files*
+   |____ InstrumentActivity.java *Activity launched for coverage. But in most cases, this should be same as MainActivity. The only difference is that it will include a `InstrumentActivityListener` to export coverage data to disk files.
+   |____ JacocoInstrumentation.java  The instrument class created by you. Also it needs to implement `InstrumentActivitylistener`.
+   |____ EndEmmaBroadCast.java A broadcast receiver which will be invoked by appium-uiautomator2-driver at the end of testing. You need implementing logic to invoke InstrumentActivityListener so as to export coverage to files.
+```
+Configure followings in your caps:  
+ `automationName` ： `uiautomator2` (case irrelevant)  
```

Comment:
Perhaps these should be an unordered list (as in the "Requirement" section above)?